src/gtimelog/main.py

```diff
@@ -806,23 +945,21 @@ def download_tasks(self):
         self.tasks_infobar_label.set_text(_("Downloading tasks..."))
         self.tasks_infobar.show()
         log.debug("Downloading tasks from %s", url)
-        cancellable = Gio.Cancellable()
-        gfile = Gio.File.new_for_uri(url)
-        self._download = (gfile, cancellable, url)
-        gfile.load_contents_async(cancellable, self.tasks_downloaded, cache_filename)
+        message = Soup.Message.new('GET', url)
+        soup_session.queue_message(message, self.tasks_downloaded, cache_filename)
```

Comment:
You're not setting `self._download` any more, which means several downloads might get started in parallel, if the server is slow and the user hits Refresh tasks several times.

That's not necessarily a problem, but I want to think about it a bit more.

Reply:
Oops.